test/runtests.jl

```diff
@@ -65,6 +65,20 @@ end
   @test stats.primal_feas ≈ 0.0
   # @test stats.dual_feas ≈ 4.63
 
+  # Test JSO-style callback signature: (nlp, solver, stats) -> Bool
+  function jso_callback(nlp_in, solver_in, stats_in)
+    @test typeof(nlp_in) <: AbstractNLPModel
+    @test hasproperty(stats_in, :iter)
+    # stop after 5 iterations
+    println("iter=", stats_in.iter, " x=", solver_in.x)
+    return stats_in.iter < 5
+  end
+  nlp = ADNLPModel(x -> (x[1] - 1)^2 + 100 * (x[2] - x[1]^2)^2, [-1.2; 1.0])
+  stats = ipopt(nlp, tol = 1e-12, callback = jso_callback, print_level = 0)
+  @test stats.status == :user
+  @test stats.solver_specific[:internal_msg] == :User_Requested_Stop
+  @test stats.iter == 5
```

Comment:
Remove comments and encapsulate this in a @testset  env. Remove println from tests.